packages/std/src/ado_base/permissioning.rs

```diff
@@ -115,51 +150,98 @@ impl LocalPermission {
 
     pub fn get_expiration(&self, env: Env) -> MillisecondsExpiration {
         match self {
-            Self::Blacklisted(expiration) => {
+            Self::Blacklisted { expiration, .. } => {
                 expiration.clone().unwrap_or_default().get_time(&env.block)
             }
             Self::Limited { expiration, .. } => {
                 expiration.clone().unwrap_or_default().get_time(&env.block)
             }
-            Self::Whitelisted(expiration) => {
+            Self::Whitelisted { expiration, .. } => {
                 expiration.clone().unwrap_or_default().get_time(&env.block)
             }
         }
     }
 
+    pub fn get_start_time(&self, env: Env) -> MillisecondsExpiration {
+        match self {
+            Self::Blacklisted { start, .. } => {
+                start.clone().unwrap_or_default().get_time(&env.block)
+            }
+            Self::Limited { start, .. } => start.clone().unwrap_or_default().get_time(&env.block),
+            Self::Whitelisted { start, .. } => {
+                start.clone().unwrap_or_default().get_time(&env.block)
+            }
+        }
+    }
+
     pub fn consume_use(&mut self) -> Result<(), ContractError> {
         if let Self::Limited { uses, .. } = self {
             *uses = uses.saturating_sub(1);
         }
 
         Ok(())
     }
+
+    pub fn validate_times(&self, env: &Env) -> Result<(), ContractError> {
+        let (start, expiration) = match self {
+            Self::Blacklisted { start, expiration }
+            | Self::Limited {
+                start, expiration, ..
+            }
+            | Self::Whitelisted { start, expiration } => (start, expiration),
+        };
+
+        if let (Some(start), Some(expiration)) = (start, expiration) {
+            let start_time = start.get_time(&env.block);
+            let exp_time = expiration.get_time(&env.block);
+
+            // Check if start time is after current time
+            if start_time.is_expired(&env.block) {
+                return Err(ContractError::StartTimeInThePast {
+                    current_time: env.block.time.seconds(),
+                    current_block: env.block.height,
+                });
+            }
+
+            // Check if expiration time is after current time
+            if exp_time.is_expired(&env.block) {
+                return Err(ContractError::ExpirationInPast {});
+            }
+
+            // Check if start time is before expiration time
+            if start_time > exp_time {
```

Comment:
This is the wrong way right? `start_time < exp_time`

Reply:
We should add test cases for this.

Reply:
I guess the comment's confusing, it returns an error if the start time is greater than the expiration time, which is the expected behavior. I've added test cases